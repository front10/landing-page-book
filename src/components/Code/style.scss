:root {
    --CodeMirror__lines_backgroundColor: #fff;
    --CodeMirror__lines_backgroundColor-dark: #dcdcdc;
    --CodeMirror__header_with: 100%;
    --CodeMirror__header_height: 45px;
    --CodeMirror__header_backgroundColor: #dcdcdc;
    --CodeMirror__footer_with: 100%;
    --CodeMirror__footer_height: 15px;
    --CodeMirror__footer_backgroundColor: #dcdcdc;
    --CodeMirror__headerDark_with: 100%;
    --CodeMirror__headerDark_height: 45px;
    --CodeMirror__headerDark_backgroundColor: rgb(68, 68, 68);
    --CodeMirror__headerDark_Color: #fff;
    --CodeMirror__footerDark_with: 100%;
    --CodeMirror__footerDark_height: 15px;
    --CodeMirror__footerDark_backgroundColor: rgb(68, 68, 68);
    --CodeMirror__footerDark_Color: #fff;
    --CodeMirror__header_copybtn_margin: 10px -10px 0 0;
    --CodeMirror__header_deletebtn_margin: 10px 0 0 0;
    --CodeMirror__height: 250px;
    --CodeMirror__font_family: Lato;
}

.CodeMirror__header {
    width: var(--CodeMirror__header_with);
    height: var(--CodeMirror__header_height);
    background-color: var(--CodeMirror__header_backgroundColor) !important;
}

.CodeMirror__footer {
    width: var(--CodeMirror__footer_with);
    height: var(--CodeMirror__footer_height);
    background-color: var(--CodeMirror__footer_backgroundColor) !important;
}

.CodeMirror__header-dark {
    width: var(--CodeMirror__headerDark_with);
    height: var(--CodeMirror__headerDark_height);
    color: var(--CodeMirror__headerDark_Color);
    background-color: var(--CodeMirror__headerDark_backgroundColor);
    .CodeMirror_btn {
        font-weight: 600;
        color: white;
    }
}

.CodeMirror__footer-dark {
    width: var(--CodeMirror__footerDark_with);
    height: var(--CodeMirror__footerDark_height);
    background-color: var(--CodeMirror__footerDark_backgroundColor);
    color: var(--CodeMirror__footerDark_Color);
}

.CodeMirror__header_copybtn {
    margin: var(--CodeMirror__header_copybtn_margin);
}

.CodeMirror__header_deletebtn {
    margin: var(--CodeMirror__header_deletebtn_margin);
}

.CodeMirror {
    height: var(--CodeMirror__height) !important;
}

.CodeMirror_btn {
    font-weight: 400;
    background-color: transparent;
}