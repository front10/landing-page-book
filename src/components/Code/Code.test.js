import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Code from './Code';

configure({ adapter: new Adapter() });

describe('Code', () => {
  let props;
  let mounted;
  const code = () => {
    if (!mounted) {
      mounted = shallow(<Code {...props} />);
    }
    return mounted;
  };
  describe('Code', () => {
    beforeEach(() => {
      props = {};
      mounted = undefined;
    });

    it('should render', () => {
      expect(shallow(<Code {...props} />)).toMatchSnapshot();
    });

    it('always renders a div', () => {
      const divs = code().find('div');

      expect(divs.length).toBeGreaterThan(0);
    });

    it('copy to clipboard', () => {
      code().instance().copyToClipboard();
      expect(code().state("scopied")).toEqual(true);
    });
  });
});
