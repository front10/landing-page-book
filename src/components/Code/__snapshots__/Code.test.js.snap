// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Code should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Code
    bgColorDark={false}
    code="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
    languageCode="javascript"
    lineNumbers={true}
    readOnly={false}
    showfooter={true}
    showheader={true}
    theme="idea"
    updateCode={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar
          className="CodeMirror__header"
          companyLink=""
          companyLogo=""
          companyName=""
          expand=""
          fixed={false}
          leftItems={Array []}
          onItemClick={[Function]}
          rightItems={Array []}
          transparent={false}
        >
          <NavbarNav
            alignItems="right"
          >
            <CopyToClipboard
              onCopy={[Function]}
              options={undefined}
              text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
            >
              <Button
                ariaLabel=""
                className="btn  CodeMirror_btn"
                disabled={false}
                loading={false}
                onClick={[Function]}
                tooltip="Copy"
                type="button"
              >
                <Icon
                  className="CodeMirror__header_copybtn"
                  icon="fa fa-clone"
                  role="link"
                  style={Object {}}
                />
              </Button>
            </CopyToClipboard>
            <Button
              ariaLabel=""
              className="btn  CodeMirror_btn"
              disabled={false}
              loading={false}
              onClick={[Function]}
              tooltip="Clear"
              type="button"
            >
              <Icon
                className="CodeMirror__header_deletebtn"
                icon="fa fa-trash-o"
                style={Object {}}
              />
            </Button>
          </NavbarNav>
        </Navbar>,
        <Unknown
          autoFocus={true}
          className=""
          onChange={[Function]}
          options={
            Object {
              "lineNumbers": true,
              "mode": "javascript",
              "readOnly": false,
              "theme": "idea",
            }
          }
          preserveScrollPosition={false}
          value="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
        />,
        <Navbar
          className="CodeMirror__header"
          companyLink=""
          companyLogo=""
          companyName=""
          expand=""
          fixed={false}
          leftItems={Array []}
          onItemClick={[Function]}
          rightItems={Array []}
          transparent={false}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <NavbarNav
            alignItems="right"
          >
            <CopyToClipboard
              onCopy={[Function]}
              options={undefined}
              text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
            >
              <Button
                ariaLabel=""
                className="btn  CodeMirror_btn"
                disabled={false}
                loading={false}
                onClick={[Function]}
                tooltip="Copy"
                type="button"
              >
                <Icon
                  className="CodeMirror__header_copybtn"
                  icon="fa fa-clone"
                  role="link"
                  style={Object {}}
                />
              </Button>
            </CopyToClipboard>
            <Button
              ariaLabel=""
              className="btn  CodeMirror_btn"
              disabled={false}
              loading={false}
              onClick={[Function]}
              tooltip="Clear"
              type="button"
            >
              <Icon
                className="CodeMirror__header_deletebtn"
                icon="fa fa-trash-o"
                style={Object {}}
              />
            </Button>
          </NavbarNav>,
          "className": "CodeMirror__header",
          "companyLink": "",
          "companyLogo": "",
          "companyName": "",
          "expand": "",
          "fixed": false,
          "leftItems": Array [],
          "onItemClick": [Function],
          "rightItems": Array [],
          "transparent": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alignItems": "right",
            "children": Array [
              <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
              >
                <Button
                  ariaLabel=""
                  className="btn  CodeMirror_btn"
                  disabled={false}
                  loading={false}
                  onClick={[Function]}
                  tooltip="Copy"
                  type="button"
                >
                  <Icon
                    className="CodeMirror__header_copybtn"
                    icon="fa fa-clone"
                    role="link"
                    style={Object {}}
                  />
                </Button>
              </CopyToClipboard>,
              <Button
                ariaLabel=""
                className="btn  CodeMirror_btn"
                disabled={false}
                loading={false}
                onClick={[Function]}
                tooltip="Clear"
                type="button"
              >
                <Icon
                  className="CodeMirror__header_deletebtn"
                  icon="fa fa-trash-o"
                  style={Object {}}
                />
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  ariaLabel=""
                  className="btn  CodeMirror_btn"
                  disabled={false}
                  loading={false}
                  onClick={[Function]}
                  tooltip="Copy"
                  type="button"
                >
                  <Icon
                    className="CodeMirror__header_copybtn"
                    icon="fa fa-clone"
                    role="link"
                    style={Object {}}
                  />
                </Button>,
                "onCopy": [Function],
                "options": undefined,
                "text": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "ariaLabel": "",
                  "children": <Icon
                    className="CodeMirror__header_copybtn"
                    icon="fa fa-clone"
                    role="link"
                    style={Object {}}
                  />,
                  "className": "btn  CodeMirror_btn",
                  "disabled": false,
                  "loading": false,
                  "onClick": [Function],
                  "tooltip": "Copy",
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "CodeMirror__header_copybtn",
                    "icon": "fa fa-clone",
                    "role": "link",
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "ariaLabel": "",
                "children": <Icon
                  className="CodeMirror__header_deletebtn"
                  icon="fa fa-trash-o"
                  style={Object {}}
                />,
                "className": "btn  CodeMirror_btn",
                "disabled": false,
                "loading": false,
                "onClick": [Function],
                "tooltip": "Clear",
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "CodeMirror__header_deletebtn",
                  "icon": "fa fa-trash-o",
                  "style": Object {},
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "className": "",
          "onChange": [Function],
          "options": Object {
            "lineNumbers": true,
            "mode": "javascript",
            "readOnly": false,
            "theme": "idea",
          },
          "preserveScrollPosition": false,
          "value": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
        },
        "ref": "editor",
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": null,
          "className": "CodeMirror__header",
          "companyLink": "",
          "companyLogo": "",
          "companyName": "",
          "expand": "",
          "fixed": false,
          "leftItems": Array [],
          "onItemClick": [Function],
          "rightItems": Array [],
          "transparent": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar
            className="CodeMirror__header"
            companyLink=""
            companyLogo=""
            companyName=""
            expand=""
            fixed={false}
            leftItems={Array []}
            onItemClick={[Function]}
            rightItems={Array []}
            transparent={false}
          >
            <NavbarNav
              alignItems="right"
            >
              <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
              >
                <Button
                  ariaLabel=""
                  className="btn  CodeMirror_btn"
                  disabled={false}
                  loading={false}
                  onClick={[Function]}
                  tooltip="Copy"
                  type="button"
                >
                  <Icon
                    className="CodeMirror__header_copybtn"
                    icon="fa fa-clone"
                    role="link"
                    style={Object {}}
                  />
                </Button>
              </CopyToClipboard>
              <Button
                ariaLabel=""
                className="btn  CodeMirror_btn"
                disabled={false}
                loading={false}
                onClick={[Function]}
                tooltip="Clear"
                type="button"
              >
                <Icon
                  className="CodeMirror__header_deletebtn"
                  icon="fa fa-trash-o"
                  style={Object {}}
                />
              </Button>
            </NavbarNav>
          </Navbar>,
          <Unknown
            autoFocus={true}
            className=""
            onChange={[Function]}
            options={
              Object {
                "lineNumbers": true,
                "mode": "javascript",
                "readOnly": false,
                "theme": "idea",
              }
            }
            preserveScrollPosition={false}
            value="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
          />,
          <Navbar
            className="CodeMirror__header"
            companyLink=""
            companyLogo=""
            companyName=""
            expand=""
            fixed={false}
            leftItems={Array []}
            onItemClick={[Function]}
            rightItems={Array []}
            transparent={false}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <NavbarNav
              alignItems="right"
            >
              <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
              >
                <Button
                  ariaLabel=""
                  className="btn  CodeMirror_btn"
                  disabled={false}
                  loading={false}
                  onClick={[Function]}
                  tooltip="Copy"
                  type="button"
                >
                  <Icon
                    className="CodeMirror__header_copybtn"
                    icon="fa fa-clone"
                    role="link"
                    style={Object {}}
                  />
                </Button>
              </CopyToClipboard>
              <Button
                ariaLabel=""
                className="btn  CodeMirror_btn"
                disabled={false}
                loading={false}
                onClick={[Function]}
                tooltip="Clear"
                type="button"
              >
                <Icon
                  className="CodeMirror__header_deletebtn"
                  icon="fa fa-trash-o"
                  style={Object {}}
                />
              </Button>
            </NavbarNav>,
            "className": "CodeMirror__header",
            "companyLink": "",
            "companyLogo": "",
            "companyName": "",
            "expand": "",
            "fixed": false,
            "leftItems": Array [],
            "onItemClick": [Function],
            "rightItems": Array [],
            "transparent": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alignItems": "right",
              "children": Array [
                <CopyToClipboard
                  onCopy={[Function]}
                  options={undefined}
                  text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                >
                  <Button
                    ariaLabel=""
                    className="btn  CodeMirror_btn"
                    disabled={false}
                    loading={false}
                    onClick={[Function]}
                    tooltip="Copy"
                    type="button"
                  >
                    <Icon
                      className="CodeMirror__header_copybtn"
                      icon="fa fa-clone"
                      role="link"
                      style={Object {}}
                    />
                  </Button>
                </CopyToClipboard>,
                <Button
                  ariaLabel=""
                  className="btn  CodeMirror_btn"
                  disabled={false}
                  loading={false}
                  onClick={[Function]}
                  tooltip="Clear"
                  type="button"
                >
                  <Icon
                    className="CodeMirror__header_deletebtn"
                    icon="fa fa-trash-o"
                    style={Object {}}
                  />
                </Button>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Button
                    ariaLabel=""
                    className="btn  CodeMirror_btn"
                    disabled={false}
                    loading={false}
                    onClick={[Function]}
                    tooltip="Copy"
                    type="button"
                  >
                    <Icon
                      className="CodeMirror__header_copybtn"
                      icon="fa fa-clone"
                      role="link"
                      style={Object {}}
                    />
                  </Button>,
                  "onCopy": [Function],
                  "options": undefined,
                  "text": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "ariaLabel": "",
                    "children": <Icon
                      className="CodeMirror__header_copybtn"
                      icon="fa fa-clone"
                      role="link"
                      style={Object {}}
                    />,
                    "className": "btn  CodeMirror_btn",
                    "disabled": false,
                    "loading": false,
                    "onClick": [Function],
                    "tooltip": "Copy",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "CodeMirror__header_copybtn",
                      "icon": "fa fa-clone",
                      "role": "link",
                      "style": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "ariaLabel": "",
                  "children": <Icon
                    className="CodeMirror__header_deletebtn"
                    icon="fa fa-trash-o"
                    style={Object {}}
                  />,
                  "className": "btn  CodeMirror_btn",
                  "disabled": false,
                  "loading": false,
                  "onClick": [Function],
                  "tooltip": "Clear",
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "CodeMirror__header_deletebtn",
                    "icon": "fa fa-trash-o",
                    "style": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "className": "",
            "onChange": [Function],
            "options": Object {
              "lineNumbers": true,
              "mode": "javascript",
              "readOnly": false,
              "theme": "idea",
            },
            "preserveScrollPosition": false,
            "value": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
          },
          "ref": "editor",
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": null,
            "className": "CodeMirror__header",
            "companyLink": "",
            "companyLogo": "",
            "companyName": "",
            "expand": "",
            "fixed": false,
            "leftItems": Array [],
            "onItemClick": [Function],
            "rightItems": Array [],
            "transparent": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
